framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -171,6 +171,10 @@ protected V getValueFromFactory(Tree tree, Node node) {
         Tree preTree = analysis.getCurrentTree();
         Pair<Tree, AnnotatedTypeMirror> preCtxt = factory.getVisitorState().getAssignmentContext();
         analysis.setCurrentTree(tree);
+        TreePath previousTreePath = factory.getVisitorState().getPath();
+        if (node instanceof MethodInvocationNode) {
```

Comment:
Can you add a comment why this is only needed for MethodInvocationNode? The assignment context also seems to be set for field updates; should the path also be set?

Reply:
(Comment added below this line.)

Reply:
@bitterfox When you look at #1520, can you undo this change?